components/tabs/src/storage.rs

```diff
@@ -273,20 +223,11 @@ impl TabsStorage {
                     last_modified,
                     remote_client,
                     record,
-                )
+                ));
             } else {
-                // A record with a device that's not in our remote clients seems unlikely, but
-                // could happen - in most cases though, it will be due to a disconnected client -
-                // so we really should consider just dropping it? (Sadly though, it does seem
-                // possible it's actually a very recently connected client, so we keep it)
-                // We should get rid of this eventually - https://github.com/mozilla/application-services/issues/5199
-                info!(
-                    "Storing tabs from a client that doesn't appear in the devices list: {}",
-                    id,
-                );
-                ClientRemoteTabs::from_record(id, last_modified, record)
+                // As per https://github.com/mozilla/application-services/issues/5199, we don't want to store these.
+                warn!("Dropping tabs from an unknown client: {id}");
```

Comment:
We'll have to make sure we monitor this a little more closely for a little incase there was some unknown-side-effects.